java/src/jmri/jmrit/logix/Tracker.java

```diff
@@ -140,103 +151,95 @@ protected String getStatus() {
-                           PathSet setB = PathSet.NO; // for paths between darkBlock and blkB
-                           if (!getPathsSet(blkA, portal).isEmpty()) {
-                               setA =  PathSet.PARTIAL;
-                               if (!getPathsSet(block, portal).isEmpty()) {
-                                   setA = PathSet.SET;
-                               }
-                           } else if (!getPathsSet(block, portal).isEmpty()) {
-                               setA = PathSet.PARTIAL;
-                           }
-                               
-                           if (!getPathsSet(blkB, port).isEmpty()) {
-                               setB =  PathSet.PARTIAL;
-                               if (!getPathsSet(block, port).isEmpty()) {
-                                   setB = PathSet.SET;
-                               }
-                           } else if (!getPathsSet(block, port).isEmpty()) {
-                               setB = PathSet.PARTIAL;
-                           }
-                           switch (setA) {
-                               case SET:
-                                   if (setB == PathSet.SET) {
-                                       darkPathSet = PathSet.SET;
-                                       _darkBlock = block;
-                                       break;
-                                   } else if (setB == PathSet.PARTIAL) {
-                                       if (darkPathSet != PathSet.SET) {
-                                           darkPathSet = PathSet.PARTIAL;
-                                           _darkBlock = block;
-                                       }
-                                   }
-                                   break;
-                               case PARTIAL:
-                                   if (setB == PathSet.SET || setB == PathSet.PARTIAL) {
-                                       if (darkPathSet != PathSet.SET) {
-                                           darkPathSet = PathSet.PARTIAL;
-                                           _darkBlock = block;
-                                       }
-                                   }
-                                   break;
-                               default:
-                                   darkBlock = null;
-                                   break;    
-                           }                           
-                       }
-                   }
-               }
+       if (pathset != PathSet.SET) {
+           if (hasExitA || hasEnterB) {
+               pathset = PathSet.PARTIAL;
            }
        }
-       if (darkPathSet == PathSet.SET) {
-           hasPathSet = darkPathSet;
-       } else if (hasPathSet == PathSet.PARTIAL) {
-           _darkBlock = darkBlock;
+       if (adjacentBlock || !recurse) {
+           /*if (log.isDebugEnabled()) {
+               log.debug("hasPathBetween: {} path for \"{}\"--\"{}\"",
```

Comment:
Please do not comment out the debugging statements.  In addition to code climate flagging this as a code smell ( because ode is commented out ) commenting out the debugging statements makes it much harder for someone else to turn them on later.

The right way to control these is in the lcf files.

Also, unless you are doing an expensive operation, the if(log.isDebugEnabled()) is not necessary.